lib/ukconsole/console.c

```diff
@@ -0,0 +1,113 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/essentials.h>
+#include <uk/arch/types.h>
+#include <uk/list.h>
+#include <uk/console.h>
+
+/* List of dynamically registered devices */
+static UK_LIST_HEAD(uk_console_device_list);
+static __u16 uk_console_device_count;
+
+static int uk_console_set_standard_once;
+
+struct uk_console *uk_console_get(__u16 id)
+{
+	struct uk_console *dev = __NULL;
+
+	uk_list_for_each_entry(dev, &uk_console_device_list, _list) {
+		if (dev->id == id)
+			return dev;
+	}
+
+	return __NULL;
+}
+
+__u16 uk_console_get_device_count(void)
+{
+	return uk_console_device_count;
+}
+
+__ssz uk_console_out(const char *buf, __sz len)
+{
+	struct uk_console *dev = __NULL;
+
+	if (!len)
+		return 0;
+
+	if (!buf)
```

Comment:
```c
if unlikely(!buf)
```
Also in error conditions you should return values from errno, i.e. in this case `-EINVAL`.